# gnomAD_v2_local
How to retrieve VCF file from gnomAD v2 lift-over in a local computer.

According what has been said in the [previous code](https://github.com/SRsina/gnomAD_in_Terra.bio/tree/main) there are the bariers when you intend to retrieve varints from gnomAD data sets and if you check the previous code there is file in its read_me file that I explaned in details so here it is intended to retrive vcf files from gnomad version 2 lift_over which is more convinient that the latest build version (version 3) as it has more data than the other if you are working with exome. 
The reason behind is that the version 2 is consist of two part:
1) exome sequencing (over 125,000 individual)
2) genome sequencing  (over 23,000 individual


but version 3 data are whole genome sequencing which is over 79,000 individual so you can decide to work with which but here we intend to work with version 2 but the lift_over version which is the same data as before but the data is mapped into build_version 38.
